power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -363,18 +346,20 @@ class MainModelImpl {
         // prepare
         auto const& input = [this, &logger, prepare_input_ = std::forward<PrepareInputFn>(prepare_input)] {
             Timer const timer{logger, LogEvent::prepare};
-            prepare_solvers<sym>();
-            assert(is_topology_up_to_date_ && is_parameter_up_to_date<sym>());
-            return prepare_input_(n_math_solvers_);
+            assert(construction_complete_);
+            prepare_solvers<sym>(state_, solver_preparation_context_, solvers_cache_status);
+            assert(
+                (solvers_cache_status.is_topology_valid() && solvers_cache_status.template is_parameter_valid<sym>()));
```

Comment:
please split up the assert

```suggestion
            assert(solvers_cache_status.is_topology_valid());
            assert(solvers_cache_status.template is_parameter_valid<sym>());
```